### PR TITLE
Add XGBRanker as a supported operator

### PR DESCRIPTION
**Pull request by KranthiGV**
This addresses #174 

In this PR:
- Added XGBoost ranker (Sklearn API) as a supported operator.
- Also, added a couple of tests around the same.

### CODE REVIEW

File: hummingbird/ml/operator_converters/xgb.py [hunk 1]
```diff
@@ -103,6 +103,23 @@ def convert_sklearn_xgb_classifier(operator, device, extra_config):
     return convert_gbdt_classifier_common(tree_infos, _get_tree_parameters, n_features, n_classes, extra_config=extra_config)
 
 
+def convert_sklearn_xgb_ranker(operator, device, extra_config):
```
Comment (interesaaat) [inline hummingbird/ml/operator_converters/xgb.py]: I think that we can just call `convert_sklearn_xgb_regressor` from `register_converter`.

**Review by interesaaat — COMMENTED** [review]
Thanks for fixing the typos!

File: hummingbird/ml/operator_converters/xgb.py (hunk 1 above)
Comment (KranthiGV) [reply hummingbird/ml/operator_converters/xgb.py]: I thought it'd be easier to read this way. Anyways, let me change that.

**Comment by KranthiGV** [comment]
I think it looks good now. Please have a look!

**Review by interesaaat — APPROVED** [review]

**Pull request merged**